app/Repository/PostRepository.php

```diff
@@ -43,7 +44,7 @@ public function getBySlug(string $slug): Post
             }
         }
 
-        throw new ShouldNotHappenException(sprintf('Post for slug "%s" was not found.', $slug));
+        throw new NotFoundHttpException(sprintf('Post for slug "%s" was not found.', $slug));
```

Comment:
to show 404 message per `NotFoundHttpException` message provided, it may require custom handler, it currently static message 

https://github.com/TomasVotruba/tomasvotruba.com/blob/d9fe7f5ea1655c51fb8c6b245a06e45b84c215ed/resources/views/errors/404.blade.php#L8